.github/workflows/validate.yml

```diff
@@ -0,0 +1,106 @@
+name: Test website
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build Jekyll site for testing
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Build the site in the Jekyll/builder container
+      run: |
+        docker run \
+        -v ${{ github.workspace }}:/srv/jekyll -v ${{ github.workspace }}/_site:/srv/jekyll/_site \
+        jekyll/builder:latest /bin/bash -c "chmod -R 777 /srv/jekyll && jekyll build --future"
+
+    - name: List result of Jekyll build
+      run: |
+        ls _site/ -l
+        ls _site/assets -l
+
+    # - name: Copy assets
+    # run: cp -r assets _site/assets
+
+    - name: Publish built site
+      uses: actions/upload-artifact@v2
+      with:
+        name: Built site ${{ github.run_number }}
+        path: ./_site
+        if-no-files-found: error
+
+  validate:
+
+    runs-on: ubuntu-latest
+    name: Validate HTML
+    needs: [build]
+
+    steps:
+    - name: Fetch built site
+      uses: actions/download-artifact@v2
+      with:
+        name: Built site ${{ github.run_number }}
+        path: ./_site
+
+    # just to satisfy the `Cyb3r-Jak3/html5validator-action` action
+    - name: Create dummy git repository
+      run: git init
+
+    - name: HTML5 Validator
+      uses: Cyb3r-Jak3/html5validator-action@44696509d19bec6bd00e5ebf29bbeda320562aac
+      with:
+        root: _site/
+
+  check-links:
+
+    runs-on: ubuntu-latest
+    name: Check links
+    needs: [build]
+
+    steps:
+    - name: Fetch built site
+      uses: actions/download-artifact@v2
+      with:
+        name: Built site ${{ github.run_number }}
+        path: ./_site
+
+    # TODO: we are not checking absolute links as pytest plugins does not support them
+    - name: Check links
+      run: |
+        pip install pytest-check-links
+        pytest _site/ --check-links \
```

Comment:
I had no idea there was a pytest link checker - that is cool!

Reply:
It's homegrown: https://github.com/jupyterlab/pytest-check-links